### DISCUSSION TITLE
Suggestion for Adding Fail2Ban Setup for Users Behind a Reverse Proxy to the Jellyfin Documentation

### DISCUSSION THREAD

**Discussion by famesjranko** [jellyfin/jellyfin#12730] [General] [answered]
Hey Jellyfin team,

I wanted to suggest adding a setup guide to the Jellyfin documentation that covers using Fail2Ban for users who are running Jellyfin behind a reverse proxy on a separate upstream server. In my current setup, I have Jellyfin running in a Docker container on a local server, but it's accessible to the internet through an upstream reverse proxy server. 

In my experience, trying to block IPs with Fail2Ban on the Jellyfin host didn’t work effectively because Docker’s internal networking can bypass the host’s `iptables` rules, allowing banned IPs to still reach Jellyfin.

So I set up Fail2Ban on the Jellyfin host to automatically ban IP addresses on the upstream server using `iptables` whenever unauthorised login attempts are detected.

I’ve put together a basic guide that walks through the setup: [Fail2Ban Upstream Proxy Chain Setup Guide](https://github.com/famesjranko/server-scripts/blob/main/fail2ban_upstream_proxy_chain_setup_guide.md).

Here’s an overview of my setup:
```
internet -> upstream-proxy <- (ban/unban IP commands) <- Fail2Ban (monitors logs)
```
```
internet -> upstream-proxy -> (allowed traffic) -> host(docker[jellyfin])
```

### What I Did:
- **Fail2Ban Configuration:** Fail2Ban is installed on the Jellyfin host and monitors the Jellyfin logs for failed login attempts or other triggers.
- **Upstream Proxy Banning:** Instead of banning IPs directly on the Jellyfin host, Fail2Ban executes `iptables` commands via SSH on the upstream reverse proxy server. This ensures that banned IPs are blocked before they reach the Jellyfin host, adding an extra layer of protection.
- **Flexible Approach:** I implemented two options for managing IP bans:
  1. **Dynamic Chains:** Fail2Ban creates and manages individual `iptables` chains for each jail on the upstream server, allowing for more granular control of IP blocking by service.
  2. **Communal Chain:** All Fail2Ban jails share a single `iptables` chain on the upstream proxy, simplifying the management of banned IPs.

### Why This Is Useful:
- **Enhanced Security:** This setup can be especially helpful for users running Jellyfin behind a reverse proxy, as it blocks malicious traffic at the proxy level before it reaches the local network.
- **Scalable:** This method works well for users with multiple services running behind the same reverse proxy, not just Jellyfin, as Fail2Ban can be configured to manage bans for all services centrally.
- **Adaptable:** The guide includes dynamic chain management, allowing advanced users to customise their ban rules based on individual services, or use a communal approach for simplicity.

I believe this setup could be valuable for other Jellyfin users with similar reverse proxy configurations and would be a useful addition to the documentation.

Let me know if you need more details, and I’d be happy to provide further insights or examples from my configuration!

**Comment by TheMelmacian** [comment, marked as answer]
You could open a PR for your guide to to add it directly to the jellyfin documentation: https://github.com/jellyfin/jellyfin.org
If they see a benefit in it they will merge it and add it to the site.
